File: services/E-Auction/E-Auction.Domain/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace E_Auction.Domain.Models
{
    public class Product
    {
        
        public int Id { get; set; }
        public string ProductName { get; set; }
        public string ShortDescription { get; set; }
        public string DetailedDescription { get; set; }
        public int CategoryId { get; set; }
        public int SellerId { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal StartingPrice { get; set; }
        public DateTime BidEndDate { get; set; }
        
        public virtual Category Category { get; set; }
        public virtual Seller Seller { get; set; }
        public virtual ICollection<Buyer> Buyers { get; set; }
    }
}

File: client/e-auction/src/app/shared/interfaces/product.ts
import {Creator} from './creator';
import {Bid} from './bid';
import {Comment} from './comment';

export interface Product {
  id?: number;
  productName?: string;
  shortDescription?: string | null;
  detailedDescription?: string;
  categoryId?: number;
  sellerId?: number;
  startingPrice?: string;
  bidEndDate?: string;
  category?: Category;
  seller?: Seller;
  buyers?: Buyer[];
}

export interface Category{
  id?: string;
  categoryName?: string;
}

export interface Seller{
  id?: number;
  firstName?: string;
  lastName?: string;
  address?: string;
  city?: string;
  state?: string;
  pin?: string;
  phone?: string;
  email?: string;
}

export interface Buyer{
  id?: string;
  firstName?: string;
  lastName?: string;
  address?: string;
  city?: string;
  state?: string;
  pin?: string;
  phone?: string;
  email?: string;
  bidAmount?:number;
}
